### DISCUSSION TITLE
Table row height customization

### DISCUSSION THREAD

**Discussion by Havock94** [Budibase/budibase#5437] [Ideas]
Hi, I think it would be useful to allow changing the table row height.

I found that the height is controlled by the variable 
`--row-height`

which is set on the table wrapper element.
I tried overriding the variable through the table Custom CSS, but the CSS is applied to the wrapper parent, therefore the variable is not overridden. 

![Schermata 2022-04-15 alle 10 49 54](https://user-images.githubusercontent.com/7635248/163549178-ca84ed58-599f-4b87-9710-26a7065f64a6.png)

Do you think it would be possible to allow changing the height? (By either setting the variable on the table element so that Custom CSS can be used, or supporting it in the Settings UI)

I'm making an app where I track movies and shows that I should watch, and at the moment the shows covers are quite small

![Schermata 2022-04-15 alle 10 53 14](https://user-images.githubusercontent.com/7635248/163549666-cbd48e18-0c8d-4e08-a3d5-ade179ebf4e2.png)

This is the effect I would like to achieve :)
![Schermata 2022-04-15 alle 10 53 40](https://user-images.githubusercontent.com/7635248/163549733-963771d2-4f48-45fe-891d-1aa9df7d10a6.png)

**Comment by melohagan** [comment]
Great idea! 

Thank you for the detailed description and the effort find a CSS solution. As you rightly stated, CSS is only inline at the moment.

I've created a feature request here: https://github.com/Budibase/budibase/issues/5476

**Comment by NicolasFabre81** [comment]
Hello @Havock94 , how did you manage to override the row height ?
I tried `--spectrum-table-cell-text-line-height : 60px !important;
--row-height : 60px !important;`
but it does not override the row height

**Reply by melohagan** [reply]
You can set the row height for the Grid block if that's an option? https://docs.budibase.com/docs/grid-block#row-settings

**Reply by NicolasFabre81** [reply]
I'd prefer to remain on a standard table... there should be a way using css no ..?

**Reply by melohagan** [reply]
There should be. It would need a bit of tinkering though. You can right click on the table and *inspect* to see the CSS that you would need to edit.

**Reply by Havock94** [reply]
@NicolasFabre81 I'm not following that particular project anymore so I don't remember the exact rule I used.
From what I remember, it couldn't be done from the table itself though, because the variables were defined on a wrapper up the DOM tree.

**Comment by NicolasFabre81** [comment]
Yes I'm currently tinkering, it works if I update the CSS in the browser
but not with me designer custom CSS.
Tried :

.spectrum-Table-row svelte-rzhjll{
--row-height:20px !important;
}

.spectrum-Table-row {
--row-height:20px !important;
}

--row-height:20px !important;

**Reply by melohagan** [reply]
I don't think you should be using `--row-height`. That's a CSS variable.

Did a quick test. Try this in an embed component:

```html
<style>
.spectrum-Table-cell {
  height: 20px !important;
}
</style>
```

**Reply by NicolasFabre81** [reply]
Great thanks it works !
<img width="553" alt="image" src="https://github.com/Budibase/budibase/assets/28592069/ff62b4f1-5e3b-4761-8ce8-5ecc9194d3cb">

**Reply by AlejandroFNadal** [reply]
Tiny extra thing: adding here `height: auto !important ` works even better, because it just takes whatever the height of the children of each cell is.

**Comment by melohagan** [comment]
The new table table component allows you to set a row height 👍 

https://docs.budibase.com/docs/table#row-settings

**Reply by AlejandroFNadal** [reply]
Thanks for the info
Sadly, we are not updating further than 2.17.6 for a few months. We have
cloned the VPS where our budibase instance is running and we will
eventually perform an update there, and then we will get that feature as
well.
The upgrades have proved way too unstable for us. Stuff always gets broken
on each upgrade (last time, I had to manually relink all data providers in
a ~50 page application because they all went to "None" after the update).
(Not complaining, you guys are doing a great job, I imagine it is just way
too hard to test all the possible crazy user cases)
